core/src/main/java/io/lionweb/lioncore/java/serialization/SerializationStatus.java

```diff
@@ -41,4 +51,50 @@ public void considerLanguageDuringSerialization(Consumer<Language> consumer, Lan
     consumer.accept(language);
     consideredLanguageIDs.add(language.getID());
   }
+
+  public void consider(SerializedClassifierInstance serializedClassifierInstance) {
+    consider(serializedClassifierInstance.getClassifier());
+    serializedClassifierInstance.getProperties().forEach(f -> consider(f.getMetaPointer()));
+    serializedClassifierInstance.getContainments().forEach(f -> consider(f.getMetaPointer()));
+    serializedClassifierInstance.getReferences().forEach(f -> consider(f.getMetaPointer()));
+  }
+
+  public void consider(MetaPointer metaPointer) {
+    UsedLanguage ul = new UsedLanguage(metaPointer.getLanguage(), metaPointer.getVersion());
+    if (!usedLanguages.contains(ul)) {
+      usedLanguages.add(ul);
+      serializedChunk.addLanguage(ul);
+    }
+  }
+
+  /** Consider the given classifier and all of its properties to track the used languages. */
+  public void consider(Classifier<?> classifier, Consumer<Language> languageConsumer) {
+    if (!hasConsideredClassifier(classifier.getID())) {
+      Objects.requireNonNull(classifier, "A node should have a concept in order to be serialized");
+      Language language = classifier.getLanguage();
+      if (language == null) {
+        throw new NullPointerException(
+            "A Concept should be part of a Language in order to be serialized. Concept "
+                + classifier
+                + " is not");
+      }
+      considerLanguageDuringSerialization(languageConsumer, language);
+      List<Feature<?>> features = classifier.allFeatures();
+      features.forEach(
+          f -> considerLanguageDuringSerialization(languageConsumer, f.getDeclaringLanguage()));
+      features.stream()
+          .filter(f -> f instanceof Property)
```

Comment:
In theory, if the property is of type _StructuredDataType_, the SDT can have fields with types of other languages that would also need to be considered.

Reply:
You are right, adding it